internal/ingress/controller/store/store_test.go

```diff
@@ -605,37 +619,89 @@ func TestStore(t *testing.T) {
 	// check invalid secret (missing ca)
 }
 
-func createNamespace(clientSet *kubernetes.Clientset, t *testing.T) string {
+func createNamespace(clientSet *fake.Clientset, t *testing.T) string {
```

Comment:
This is still using a specific type, you can replace any Clientset by `kubernetes.Interface` to make it portable.

Reply:
I don't think we can cast fake.ClientSet to kubernetes.Clientset.

Reply:
Not `ClientSet`, `Interface`. All REST clients implement it, including `fake`.

Reply:
Can you provide me an example of how it should look. Having some issues trying to properly type-cast.

Reply:
Awesome Thanks, I see now:

fakeclient:
```
clientset "k8s.io/client-go/kubernetes"
clientset.Interface = &Clientset{}

# &Clientset{} is:
# type Clientset struct {
#	testing.Fake
#	discovery *fakediscovery.FakeDiscovery
#}
```

Reply:
The reason why it works is not the contents of the struct, just the [methods the struct implements](https://godoc.org/k8s.io/client-go/kubernetes/fake#Clientset.Admissionregistration) (in other words, `Admissionregistration()` ... `StorageV1beta1()`)